persistent/persistent.cabal

```diff
@@ -17,7 +17,7 @@ extra-source-files: ChangeLog.md README.md
 library
     build-depends:   
         base                     >= 4.11.1.0     && < 5
-      , aeson                    >= 1.0 && < 2.2
+      , aeson                    == 2.1.*
```

Comment:
Why is this bound so much tighter now?

Reply:
Was trying to avoid the issue with Aeson error message changes causing tests to fail.

Reply:
And of course it failed.

Reply:
The problem is that `persistent` depends on `shakespeare` which has a base dependency of `(>=4.12 && <5)` (ie only build with `ghc-8.6` and later). 

I do not see an easy solution to this other than dropping `ghc-8.4` (the version in that series was release on 14th October 2018).

Reply:
Ahhh, I see, so the solver needs to pick a newer version of `shakespeare`  to satisfy the `aeson > 2`, and *that* `shakespeare` also bumps the base bound. So `shakespeare` is blocking any package from compiling with GHC 8.4 that also supports `aeson >= 2`. That's annoying!

Testing locally, `shakespeare` HEAD builds fine with GHC 8.4. So it's not even necessary.

Reply:
So what do you propose is the solution required to get this PR merged?

Reply:
I'm OK dropping GHC 8.4 support; I'll make a follow-on issue to either restore it or finish the job (and delete relevant CPP bits for compatibility)